kernel/syscall.c

```diff
@@ -1684,6 +1684,59 @@ long myst_syscall_readlink(const char* pathname, char* buf, size_t bufsiz)
     return ret;
 }
 
+long myst_syscall_readlinkat(
+    int dirfd,
+    const char* pathname,
+    char* buf,
+    size_t bufsiz)
+{
+    long ret = 0;
+    struct locals
+    {
+        char abspath[PATH_MAX];
+    };
+    struct locals* locals = NULL;
+
+    if (!pathname || !buf || !bufsiz)
+        ERAISE(-EINVAL);
+
+    /* If pathname is absolute, then ignore dirfd */
+    if (*pathname == '/' || dirfd == AT_FDCWD)
+    {
+        ret = myst_syscall_readlink(pathname, buf, bufsiz);
+    }
+    else if (*pathname == '\0')
+    {
+        /*
+         * ATTN: Since Linux 2.6.39, pathname can be an empty string, in which
+         * case the call operates on the symbolic link referred to by dirfd.
+         * But dirfd should have been obtained using open with the O_PATH
+         * and O_NOFOLLOW flags. Our existing implementation of ext2_open()
+         * dosn't support the O_PATH flag. If the trailing component
+         * (i.e., basename) of pathname is a symbolic link, then the open
```

Comment:
Are you describing this case(from `man 2 open`) - 
```
ELOOP  pathname was a symbolic link, and flags specified O_NOFOLLOW but not O_PATH.
```
?

Reply:
Right, see my comment above.